src/kernel/init.c

```diff
@@ -223,7 +223,6 @@ closure_function(1, 3, void, attach_storage,
     } else {
         rootfs_init(0, offset, r, w, length);
     }
-    closure_finish();
```

Comment:
Yes, attach_storage is called more than once if there is more than one storage device, and this happens when multiple volumes are attached to an instance. So it should not be a no-op, because it needs to handle volume mounting.